merino-adm/src/remote_settings/mod.rs

```diff
@@ -241,12 +250,28 @@ impl RemoteSettingsSuggester {
         }
 
         // Download and process all the attachments.
-        let mut data_records = records.clone();
-        data_records.retain(|r| r.get("type") == Some(&serde_json::json!("data")));
+        let all_data_records = records.clone();
+
+        // Get records whose `type` field is either `data` or `offline-expansion-data`,
+        // prefer `offline-expansion-data` over `data` if the former is present.
+        //
+        // See details: https://mozilla-hub.atlassian.net/browse/CONSVC-1817
+        let (mut data_records, offline_data_records): (Vec<_>, Vec<_>) = all_data_records
+            .into_iter()
+            .filter(|r| {
+                matches!(
+                    r.get("type"),
+                    Some(&serde_json::Value::String(ref val)) if val == "data" || val =="offline-expansion-data"
```

Comment:
whitespace nit (cargo fmt misses it since it's a macro)
```suggestion
                    Some(&serde_json::Value::String(ref val)) if val == "data" || val == "offline-expansion-data"
```

Reply:
oops 👍